pkg/cmd/workspace/create.go

```diff
@@ -506,3 +509,57 @@ func dedupProjectNames(projects *[]apiclient.CreateProjectDTO) {
 		}
 	}
 }
+
+func GetGitProviderGpgKey(apiClient *apiclient.APIClient, ctx context.Context, activeProfile config.Profile, workspaceId string) (string, error) {
```

Comment:
I see what you were going for here but this is way too hacky 😄.

It's not intended that we extract authentication variables from the project and then construct the agent api key outside of a project.

To solve this, I recommend the following:
1. Add a `maskToken bool` argument to [GetGitProviderForUrl](https://github.com/daytonaio/daytona/blob/main/pkg/api/controllers/gitprovider/gitprovider.go#L59). If the argument is set to `true`, we should set the token to `""` before returning the object from the API.
2. Modify the registration of the controller method in the project auth group here (https://github.com/daytonaio/daytona/blob/main/pkg/api/server.go#L260) so that you pass `false` as the second argument.
3. Register the controller method in the public auth scope (similarly to how other git provider controller methods are registered) and set the second argument to `true`.
4. Regenerate the api client with `./hack/swagger.sh`
5. Now you can use `apiClient.GitProviderAPI.GetGitProviderForUrl` in this function here and not have to hack your way around.

The aim of 2. and 3. is that the user can fetch the git provider token only inside the project which is the intended way.

Additionally, this function should take `projectName` as an argument because the user can enter other projects in the workspace using `daytona ssh` and you hardcoded it to always fetch the first project in the list.


Hope this makes sense, let me know if you have any questions.

Reply:
```golang
projectGroup.POST(workspaceController.BasePath()+"/:workspaceId/:projectId/state", workspace.SetProjectState)
		projectGroup.GET(gitProviderController.BasePath()+"/for-url/:url/unmask", func(ctx *gin.Context) {
			gitprovider.GetGitProviderForUrl(ctx, false)
```
we will need to rename it and if we dont then we will get error  panic: handlers are already registered for path '/gitprovider/for-url/:url'

Reply:
```go
gitProviderController := protected.Group("/gitprovider")
	{
		gitProviderController.GET("/", gitprovider.ListGitProviders)
		gitProviderController.PUT("/", gitprovider.SetGitProvider)
		gitProviderController.DELETE("/:gitProviderId", gitprovider.RemoveGitProvider)
		gitProviderController.GET("/:gitProviderId/user", gitprovider.GetGitUser)
		gitProviderController.GET("/:gitProviderId/namespaces", gitprovider.GetNamespaces)
		gitProviderController.GET("/:gitProviderId/:namespaceId/repositories", gitprovider.GetRepositories)
		gitProviderController.GET("/:gitProviderId/:namespaceId/:repositoryId/branches", gitprovider.GetRepoBranches)
		gitProviderController.GET("/:gitProviderId/:namespaceId/:repositoryId/pull-requests", gitprovider.GetRepoPRs)
		gitProviderController.POST("/context", gitprovider.GetGitContext)
		gitProviderController.POST("/context/url", gitprovider.GetUrlFromRepository)
		gitProviderController.GET("/id-for-url/:url", gitprovider.GetGitProviderIdForUrl)
		gitProviderController.GET("/for-url/:url", func(ctx *gin.Context) {
			gitprovider.GetGitProviderForUrl(ctx, true)
		})
	}
```

Reply:
Done the required chnages.
will rename it accordingly if you say so

Reply:
Yes

Reply:
please point out line or file directly `AuthMiddleware`

Reply:
Nice work! This seems to work great when gpg is installed in the project. I'll resolve this conversation and start another one with other comments.